File: Resources/Packet/Join.cs
using System.IO;

namespace Resources.Packet {
    public class Join {
        public const int packetID = 16;

        public int unknown;
        public long guid;
        public byte[] junk;

        public Join() { }

        public Join(BinaryReader reader) {
            unknown = reader.ReadInt32();
            guid = reader.ReadInt64();
            junk = reader.ReadBytes(0x1168);
        }

        public void Write(BinaryWriter writer, bool writePacketID = true) {
            if(writePacketID) {
                writer.Write(packetID);
            }
            writer.Write(unknown);
            writer.Write(guid);
            writer.Write(junk);
        }
    }
}

File: common/src/main/kotlin/packets/Join.kt
package packets

class Join(
	val unknown: Int,
	val assignedID: CreatureID,
	val junk: ByteArray
) : Packet(Opcode.Join) {
	override suspend fun writeTo(writer: Writer) {
		writer.writeInt(unknown)
		writer.writeLong(assignedID.value)
		writer.writeByteArray(junk)
	}

	companion object {
		suspend fun readFrom(reader: Reader): Join {
			return Join(
				unknown = reader.readInt(),
				assignedID = CreatureID(reader.readLong()),
				junk = reader.readByteArray(0x1168)
			)
		}
	}
}